### ISSUE TITLE
Add an Expand All button to the references tree views

### ISSUE THREAD

**Issue by sean-mcmanus** [microsoft/vscode-cpptools#4333]
There's a collapse all button.

Adding "Expand All" to the main "References: Results" would require us to replace VS Code's tree view container for the Confirmed References.

**Comment by nyklotocky** [comment]
Agreed. Although ideally there would be a setting to enable expanding the tree by default, similar to the way the search setting does (the setting `search.collapseResults`)

**Comment by Trass3r** [comment]
> Adding "Expand All" to the main "References: Results" would require us to replace VS Code's tree view container for the Confirmed References.

vscode-clangd has the same problem. Shouldn't this be solved in vscode then by replacing the collapse button with a toggle expand/collapse button like in the search results view?

**Comment by sean-mcmanus** [comment]
@Trass3r Yes, VS Code itself could add an expand all for their References: Results UI, but our extension adds additional tree views, so this issue is mostly in regards to that. See the screenshot: 
![image](https://user-images.githubusercontent.com/19859882/90173768-ca0e8b00-dd59-11ea-904e-91aa8628f8d8.png)

Right now, the collapse button only applies to one of the tree views, so if VS Code added an expand, we would also need to add it to our "Other References Result" tree view.

**Comment by Trass3r** [comment]
Collapse All seems to be a TreeView [feature](https://github.com/microsoft/vscode-references-view/blob/846367409ec890e3415c919db772c3fbfe01f41e/src/extension.ts#L28) though. Maybe they could provide a general solution.

**Comment by hewitt-codaio** [comment]
Expand all by default would be enough for me.  This prevents me from using VS Code because I depend heavily on inspecting references.

**Comment by adam-flynn** [comment]
You can get expand all by default by changing [line 273 of vscode-references-view / `src/references/model.ts`](https://github.com/microsoft/vscode-references-view/blob/4d7b4ea8398b23e7b7c3ec2776927fcd9b35a8c5/src/references/model.ts#L273) from

    result.collapsibleState = vscode.TreeItemCollapsibleState.Collapsed;

to

    result.collapsibleState = vscode.TreeItemCollapsibleState.Expanded;

 I manually changed it in the compiled `resources/app/extensions/ms-vscode.references-view/dist/extension.js` and it worked.

**Comment by mosesyu** [comment]
Changed the compiled `resources/app/extensions/ms-vscode.references-view/dist/extension.js`, but it did not change anything?

> You can get expand all by default by changing [line 273 of vscode-references-view / `src/references/model.ts`](https://github.com/microsoft/vscode-references-view/blob/4d7b4ea8398b23e7b7c3ec2776927fcd9b35a8c5/src/references/model.ts#L273) from
> 
> ```
> result.collapsibleState = vscode.TreeItemCollapsibleState.Collapsed;
> ```
> 
> to
> 
> ```
> result.collapsibleState = vscode.TreeItemCollapsibleState.Expanded;
> ```
> 
> I manually changed it in the compiled `resources/app/extensions/ms-vscode.references-view/dist/extension.js` and it worked.
> 
> But that said, I'll seriously still like a Expand All button.

**Comment by BloodyRain2k** [comment]
> This feature request has received enough votes to be added to our backlog.

Are there any updates on whenever this extension or VSCode itself will ever change the Collapse to a Toggle?

**Comment by sean-mcmanus** [comment]
@BloodyRain2k I'm not aware of any updates.

**Comment by Hau-Hau** [comment]
Hi, I have created new issue in vscode repo and implemented setting to control default behaviour of References view.
https://github.com/microsoft/vscode/pull/197594
